pkg/scanner/scanner.go

```diff
@@ -0,0 +1,171 @@
+package scanner
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/fluxcd/pkg/version"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/go-logr/logr"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	"github.com/weaveworks/profiles/pkg/gitrepository"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+//go:generate counterfeiter -o fakes/fake_git_client.go . GitClient
+//GitClient client for interacting with git
+type GitClient interface {
+	ListTags(url string, secret *corev1.Secret) ([]string, error)
+}
+
+//go:generate counterfeiter -o fakes/fake_repo_manager.go . GitRepositoryManager
+//GitRepositoryManager for managing gitrepositorys
+type GitRepositoryManager interface {
+	CreateAndWaitForResources(repo profilesv1.Repository, tags []gitrepository.Instance) ([]*sourcev1.GitRepository, error)
+	DeleteResources([]*sourcev1.GitRepository) error
+}
+
+//go:generate counterfeiter -o fakes/fake_http_client.go . HTTPClient
+//HTTPClient for making HTTP requests
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+//Scanner for scanning repositorys
+type Scanner struct {
+	gitRepositoryManager GitRepositoryManager
+	gitClient            GitClient
+	httpClient           HTTPClient
+	logger               logr.Logger
+	namespace            string
+	ctx                  context.Context
+}
+
+//New returns a Scanner
+func New(gitRepositoryManager GitRepositoryManager, gitClient GitClient, httpClient HTTPClient, logger logr.Logger) *Scanner {
+	return &Scanner{
+		gitRepositoryManager: gitRepositoryManager,
+		gitClient:            gitClient,
+		httpClient:           httpClient,
+		logger:               logger,
+		namespace:            "profiles-system",
+		ctx:                  context.TODO(),
+	}
+}
+
+//ScanRepository for profiles
+func (s *Scanner) ScanRepository(repo profilesv1.Repository, secret *corev1.Secret) ([]profilesv1.ProfileCatalogEntry, error) {
+	tags, err := s.gitClient.ListTags(repo.URL, secret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list tags: %w", err)
+	}
+	s.logger.Info("found tags", "url", repo.URL, "tags", tags)
+
+	var instances []gitrepository.Instance
+	for _, tag := range tags {
+		semver, path := getSemverAndPathFromTag(tag)
+		_, err := version.ParseVersion(semver)
```

Comment:
do we care what happens if this err is not nil? am i missing the check for it?

Reply:
if its not valid semver, its not a tag we care about so we ignore it